src/percy-agent-client/dom.ts

```diff
@@ -0,0 +1,210 @@
+export interface DOMOptions {
+  enableJavaScript?: boolean,
+  domTransformation?: (dom: any) => void
+}
+
+/**
+ * A single class to encapsulate all DOM operations that need to be preformed to
+ * capture the customers application state.
+ *
+ */
+class DOM {
+  originalDOM: any
+  clonedDOM: any
+  options: DOMOptions
+
+  readonly defaultDoctype = '<!DOCTYPE html>'
+
+  constructor(dom: any, options?: DOMOptions) {
+    this.originalDOM = dom
+    this.options = options || {}
+    this.clonedDOM = this.cloneDOM()
+  }
+
+  /**
+   * Returns the final DOM string with all of the necessary transforms
+   * applied. This is the string that is passed to the API and then rendered by
+   * our API.
+   *
+   */
+  snapshotString(): string {
```

Comment:
I'd wondering about being more specific here `domSnapshotString` or some other name - to avoid possible confusion with Percy's Snapshots.  I.e. in the near future we think each Percy Snapshot may have multiple DOM Snapshots.

Reply:
I like the name `domSnapshotString`.

Though, it is on the `DOM` class:

```js
let dom = new DOM(document);

// this
dom.snapshotString()
// vs
dom.domSnapshotString()
```

I don't have a strong feel either way